--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -376,8 +376,8 @@ if(BUILD_TESTING)
   add_subdirectory(tests)
 endif()
 add_subdirectory(crypto)
-add_subdirectory(third-party)
 if(NOT ENABLE_LIB_ONLY)
+  add_subdirectory(third-party)
   add_subdirectory(examples)
 endif()
 
@@ -401,6 +401,7 @@ message(STATUS "summary of build options:
       Static:         ${ENABLE_STATIC_LIB}
     Test:
       Build Test:     ${BUILD_TESTING}
+    Library only:     ${ENABLE_LIB_ONLY}
     Libs:
       OpenSSL:        ${HAVE_OPENSSL} (LIBS='${OPENSSL_LIBRARIES}')
       Libev:          ${HAVE_LIBEV} (LIBS='${LIBEV_LIBRARIES}')
