--- a/CMakeLists.txt	2024-10-15 03:56:48.272443286 +0200
+++ b/CMakeLists.txt	2024-10-15 03:56:50.524459541 +0200
@@ -352,8 +352,10 @@
   add_subdirectory(tests)
 endif()
 add_subdirectory(crypto)
-add_subdirectory(third-party)
-add_subdirectory(examples)
+if (NOT ENABLE_LIB_ONLY)
+  add_subdirectory(third-party)
+  add_subdirectory(examples)
+endif()
 
 
 string(TOUPPER "${CMAKE_BUILD_TYPE}" _build_type)
@@ -376,6 +378,7 @@
       Static:         ${ENABLE_STATIC_LIB}
     Test:
       Build Test:     ${BUILD_TESTING}
+    Library only:     ${ENABLE_LIB_ONLY}
     Libs:
       OpenSSL:        ${HAVE_OPENSSL} (LIBS='${OPENSSL_LIBRARIES}')
       Libev:          ${HAVE_LIBEV} (LIBS='${LIBEV_LIBRARIES}')
--- a/CMakeOptions.txt	2024-10-05 10:27:01.000000000 +0200
+++ b/CMakeOptions.txt	2024-10-15 03:55:16.151778358 +0200
@@ -1,5 +1,6 @@
 # Features that can be enabled for cmake (see CMakeLists.txt)
 
+option(ENABLE_LIB_ONLY   "Build libngtcp2 only" OFF)
 option(ENABLE_STATIC_LIB "Build as static libraries" ON)
 option(ENABLE_SHARED_LIB "Build as shared libraries" ON)
 
